Binaryen no longer builds in emsdk CI (Ubuntu 20.04.6 LTS)

```
/root/project/binaryen/main/test/gtest/stringify.cpp: In function 'std::vector<wasm::SuffixTree::RepeatedSubstring> repeatSubstrings(std::vector<unsigned int, std::allocator<unsigned int> >)':
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: error: class template argument deduction failed:
     std::vector(st.begin(), st.end());
                                     ^
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: error: no matching function for call to 'vector(wasm::SuffixTree::iterator, wasm::SuffixTree::iterator)'
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:411:2: note: candidate: template<class _Tp, class _Alloc, class _InputIterator, class> vector(_InputIterator, _InputIterator, const _Alloc&)-> std::vector<_Tp, _Alloc>
  vector(_InputIterator __first, _InputIterator __last,
  ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:411:2: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   couldn't deduce template parameter '_Tp'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:383:7: note: candidate: template<class _Tp, class _Alloc> vector(std::initializer_list<_Tp>, const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(initializer_list<value_type> __l,
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:383:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   'wasm::SuffixTree::RepeatedSubstringIterator' is not derived from 'std::initializer_list<_Tp>'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:358:7: note: candidate: template<class _Tp, class _Alloc> vector(std::vector<_Tp, _Alloc>&&, const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(vector&& __rv, const allocator_type& __m)
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:358:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   'wasm::SuffixTree::iterator {aka wasm::SuffixTree::RepeatedSubstringIterator}' is not derived from 'std::vector<_Tp, _Alloc>'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:348:7: note: candidate: template<class _Tp, class _Alloc> vector(const std::vector<_Tp, _Alloc>&, const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(const vector& __x, const allocator_type& __a)
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:348:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   'wasm::SuffixTree::iterator {aka wasm::SuffixTree::RepeatedSubstringIterator}' is not derived from 'const std::vector<_Tp, _Alloc>'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:344:7: note: candidate: template<class _Tp, class _Alloc> vector(std::vector<_Tp, _Alloc>&&)-> std::vector<_Tp, _Alloc>
       vector(vector&& __x) noexcept
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:344:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   'wasm::SuffixTree::iterator {aka wasm::SuffixTree::RepeatedSubstringIterator}' is not derived from 'std::vector<_Tp, _Alloc>'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:326:7: note: candidate: template<class _Tp, class _Alloc> vector(const std::vector<_Tp, _Alloc>&)-> std::vector<_Tp, _Alloc>
       vector(const vector& __x)
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:326:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   'wasm::SuffixTree::iterator {aka wasm::SuffixTree::RepeatedSubstringIterator}' is not derived from 'const std::vector<_Tp, _Alloc>'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:295:7: note: candidate: template<class _Tp, class _Alloc> vector(std::vector<_Tp, _Alloc>::size_type, const value_type&, const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(size_type __n, const value_type& __value,
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:295:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:25: note:   cannot convert 'wasm::SuffixTree::begin()()' (type 'wasm::SuffixTree::iterator {aka wasm::SuffixTree::RepeatedSubstringIterator}') to type 'std::vector<_Tp, _Alloc>::size_type {aka long unsigned int}'
     std::vector(st.begin(), st.end());
                 ~~~~~~~~^~
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:283:7: note: candidate: template<class _Tp, class _Alloc> vector(std::vector<_Tp, _Alloc>::size_type, const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(size_type __n, const allocator_type& __a = allocator_type())
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:283:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   couldn't deduce template parameter '_Tp'
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:270:7: note: candidate: template<class _Tp, class _Alloc> vector(const allocator_type&)-> std::vector<_Tp, _Alloc>
       vector(const allocator_type& __a) _GLIBCXX_NOEXCEPT
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:270:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   candidate expects 1 argument, 2 provided
     std::vector(st.begin(), st.end());
                                     ^
In file included from /usr/include/c++/7/vector:64:0,
                 from /root/project/binaryen/main/src/wasm.h:34,
                 from /root/project/binaryen/main/src/ir/boolean.h:20,
                 from /root/project/binaryen/main/src/ir/bits.h:20,
                 from /root/project/binaryen/main/src/ir/properties.h:20,
                 from /root/project/binaryen/main/src/ir/iteration.h:20,
                 from /root/project/binaryen/main/src/ir/branch-utils.h:20,
                 from /root/project/binaryen/main/src/ir/utils.h:20,
                 from /root/project/binaryen/main/test/gtest/stringify.cpp:1:
/usr/include/c++/7/bits/stl_vector.h:259:7: note: candidate: template<class _Tp, class _Alloc> vector()-> std::vector<_Tp, _Alloc>
       vector()
       ^~~~~~
/usr/include/c++/7/bits/stl_vector.h:259:7: note:   template argument deduction/substitution failed:
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: note:   candidate expects 0 arguments, 2 provided
     std::vector(st.begin(), st.end());
                                     ^
/root/project/binaryen/main/test/gtest/stringify.cpp:257:37: error: expression list treated as compound expression in functional cast [-fpermissive]
At global scope:
```

--- a/test/gtest/stringify.cpp
+++ b/test/gtest/stringify.cpp
@@ -254,7 +254,7 @@ std::vector<SuffixTree::RepeatedSubstring>
 repeatSubstrings(std::vector<uint32_t> hashString) {
   SuffixTree st(hashString);
   std::vector<SuffixTree::RepeatedSubstring> substrings =
-    std::vector(st.begin(), st.end());
+    std::vector<SuffixTree::RepeatedSubstring>(st.begin(), st.end());
   std::sort(
     substrings.begin(),
     substrings.end(),

--- a/test/gtest/stringify.cpp
+++ b/test/gtest/stringify.cpp
@@ -253,8 +253,7 @@ TEST_F(StringifyTest, Stringify) {
 std::vector<SuffixTree::RepeatedSubstring>
 repeatSubstrings(std::vector<uint32_t> hashString) {
   SuffixTree st(hashString);
-  std::vector<SuffixTree::RepeatedSubstring> substrings =
-    std::vector<SuffixTree::RepeatedSubstring>(st.begin(), st.end());
+  std::vector<SuffixTree::RepeatedSubstring> substrings(st.begin(), st.end());
   std::sort(
     substrings.begin(),
     substrings.end(),

